From bc378714e86664eed5fcd3b3f987c5509736dda9 Mon Sep 17 00:00:00 2001
From: Rob Herring <r.herring@freescale.com>
Date: Tue, 1 Jul 2008 21:34:47 -0500
Subject: [PATCH] ENGR00081519 ipu3: disable FG CSC when FG is disabled

Disable the FG CSC when FG is disabled. This fixes a problem with the display
fading away and speeds up the disable time.

Signed-off-by: Rob Herring <r.herring@freescale.com>
---
 drivers/mxc/ipu3/ipu_disp.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/drivers/mxc/ipu3/ipu_disp.c b/drivers/mxc/ipu3/ipu_disp.c
index 86754d4..652a26f 100644
--- a/drivers/mxc/ipu3/ipu_disp.c
+++ b/drivers/mxc/ipu3/ipu_disp.c
@@ -466,6 +466,7 @@ void _ipu_dp_dc_disable(ipu_channel_t channel)
 	int ret;
 	uint32_t lock_flags;
 	uint32_t reg;
+	uint32_t csc;
 	uint32_t *dc_ch_conf;
 	int irq = 0;
 	int timeout = 50;
@@ -483,7 +484,12 @@ void _ipu_dp_dc_disable(ipu_channel_t channel)
 		spin_lock_irqsave(&ipu_lock, lock_flags);
 
 		reg = __raw_readl(DP_COM_CONF(DP_SYNC));
-		__raw_writel(reg & ~DP_COM_CONF_FG_EN, DP_COM_CONF(DP_SYNC));
+		csc = reg & DP_COM_CONF_CSC_DEF_MASK;
+		if (csc == DP_COM_CONF_CSC_DEF_FG)
+			reg &= ~DP_COM_CONF_CSC_DEF_MASK;
+
+		reg &= ~DP_COM_CONF_FG_EN;
+		__raw_writel(reg, DP_COM_CONF(DP_SYNC));
 
 		reg = __raw_readl(IPU_SRM_PRI2) | 0x8;
 		__raw_writel(reg, IPU_SRM_PRI2);
-- 
1.5.4.4

